readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/monitoring/PrometheusMetrics.java

```diff
@@ -4,51 +4,138 @@
+        registerPartitionCollector(CHECKPOINTS_FAILURES, eventName, String.valueOf(partition));
+        return buildChild(CHECKPOINTS_FAILURES, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child checkPointSuccessesCounter(String eventName, int partition) {
+        registerPartitionCollector(CHECKPOINTS_SUCCESSES, eventName, String.valueOf(partition));
+        return buildChild(CHECKPOINTS_SUCCESSES, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child messageWritingFailuresCounter(String eventName, int partition) {
+        registerPartitionCollector(MESSAGES_WRITING_FAILURES, eventName, String.valueOf(partition));
+        return buildChild(MESSAGES_WRITING_FAILURES, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child messageWrittenCounter(String eventName, int partition) {
+        registerPartitionCollector(MESSAGES_WRITTEN, eventName, String.valueOf(partition));
+        return buildChild(MESSAGES_WRITTEN, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child hearbeatsSentCounter(String eventName, int partition) {
+        registerPartitionCollector(HEARTBEATS_SENT, eventName, String.valueOf(partition));
+        return buildChild(HEARTBEATS_SENT, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child currentRunningOffsetsGauge(String eventName, int partition) {
+        registerPartitionCollector(CURRENT_RUNNING_OFFSETS, eventName, String.valueOf(partition));
+        return buildChild(CURRENT_RUNNING_OFFSETS, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child latestCommittedOffsetGauge(String eventName, int partition) {
+        registerPartitionCollector(LATEST_COMMITTED_OFFSETS, eventName, String.valueOf(partition));
+        return buildChild(LATEST_COMMITTED_OFFSETS, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child latestCommittedTimestampGauge(String eventName, int partition) {
+        registerPartitionCollector(LATEST_COMMITTED_TIMESTAMPS, eventName, String.valueOf(partition));
+        Gauge.Child gauge = buildChild(LATEST_COMMITTED_TIMESTAMPS, eventName, String.valueOf(partition));
+        return new Gauge.Child() {
+            @Override
+            public void set(double val) {
+                gauge.set(Double.max(gauge.get(), val));
+            }
+
+            @Override
+            public double get() {
+                return gauge.get();
+            }
+        };
+    }
+
+    public static void clearCollectors() {
+        REGISTERED_COLLECTORS.forEach((collector, labels) -> {
+            collector.clear();
+            labels.clear();
+        });
+    }
+
+    public static void clearPartitionCollectors(int partition) {
+        REGISTERED_COLLECTORS.forEach((collector, labels) -> labels
+            .stream()
+            .filter(list -> String.valueOf(partition).equals(list.get(PARTITION_LABEL_IDX)))
```

Comment:
Didn't like it either. I propose to remember the partition when registering the collector and to filter on that rather than looking in the labels at a specific id.